Source/WebCore/rendering/svg/LegacyRenderSVGForeignObject.h

```diff
@@ -60,6 +60,7 @@ class LegacyRenderSVGForeignObject final : public RenderSVGBlock {
 
     const AffineTransform& localToParentTransform() const override;
     AffineTransform localTransform() const override { return m_localTransform; }
+    bool isUnderSVGRoot() const;
```

Comment:
I wonder if we need to generalize this mechanism. What happens if you try to re-parent e.g. a <g> container to become the body element? All LegacyRenderSVG… classes contain the same assertion in layout().

Do we have other mechanisms in place that prevent to move SVG elements to form unexpected render trees? You need to duplicate your test for <svg> (should work), <g> (should still assert) and also for <use> which has a history of subtle bugs.

Reply:
> Do we have other mechanisms in place that prevent to move SVG elements to form unexpected render trees?
Based on Slack chat with a few engineers, I don't think there are things in place to do that. 

I'm not sure if we should fix all of those issues with the same patch - My intention is to have a targeted fix for the bug in question, but if we think there are other improvements that can be done, maybe those should be separate changes?

Let me try out some of the other elements to see if that increases scope by a lot.

Reply:
I tried out various other cases like:
```
<!-- <svg>
    <circle id="foreignObject" cx="5" cy="5" r="4"/>
</svg> -->

<!-- <svg>
    <path id="foreignObject"
    d="M 10,30
           A 20,20 0,0,1 50,30
           A 20,20 0,0,1 90,30
           Q 90,60 50,90
           Q 10,60 10,30 z" />
</svg> -->

<!-- <svg>
    <ellipse id="foreignObject" cx="100" cy="50" rx="100" ry="50" />
</svg> -->

<!-- <svg>
    <circle id="foreignObject" cx="5" cy="5" r="4"/>
</svg> -->

<!-- <svg>
    <rect id="foreignObject" width="100" height="100"/>
</svg> -->
```
And none of those lead to a crash, mainly because the render tree in those cases just has a top level RenderView, so it doesn't get to the layout for SVG elements. Looks like we can proceed with this change for foreignObject.